contracts/ClaimAndStake.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Multicall.sol";
+import "@uma/core/contracts/merkle-distributor/implementation/MerkleDistributorInterface.sol";
+
+interface AcceleratingDistributorInterface {
+    function stakeFor(
+        address beneficiary,
+        address stakedToken,
+        uint256 amount
+    ) external;
+}
+
+interface AcrossMerkleDistributorInterface is MerkleDistributorInterface {
+    function claimFor(MerkleDistributorInterface.Claim memory _claim) external;
+}
+
+/**
+ * @notice Allows claimer to claim tokens from AcrossMerkleDistributor and stake into AcceleratingDistributor
+ * atomically in a single transaction. This intermediary contract also removes the need for claimer to approve
+ * AcceleratingDistributor to spend its staking tokens.
+ */
+
+contract ClaimAndStake is ReentrancyGuard, Ownable, Multicall {
```

Comment:
nit: why does this contract need to be ownable?

Reply:
it no longer does! now that I've removed the `setDistributorContracts`